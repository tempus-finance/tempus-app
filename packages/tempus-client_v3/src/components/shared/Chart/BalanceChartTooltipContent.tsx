import { FC } from 'react';
import { Ticker } from 'tempus-core-services';
import FormattedDate from '../FormattedDate';
import Typography from '../Typography';
import './Chart.scss';

interface BalanceChartTooltipContentProps {
  title: string;
  value: string;
  currency: Ticker;
  date: Date;
}

const BalanceChartTooltipContent: FC<BalanceChartTooltipContentProps> = props => {
  const { title, value, currency, date } = props;

  return (
    <div className="tc__chart__balance-tooltip-content">
      <Typography variant="body-primary" weight="bold" color="text-secondary">
        {title}
      </Typography>
      <div className="tc__chart__balance-tooltip-content-value">
        <Typography variant="subtitle" type="mono" weight="bold">
          {value} {currency}
        </Typography>
      </div>
      <FormattedDate date={date} size="small" textColor="text-secondary" separatorContrast="low" />
    </div>
  );
};

export default BalanceChartTooltipContent;
