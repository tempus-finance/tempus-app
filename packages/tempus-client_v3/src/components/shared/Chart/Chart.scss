.tc__chart__date-tick-container {
  $date-tick-width: 40px;
  $date-tick-height: 32px;

  --tick-width: #{$date-tick-width};
  --tick-height: #{$date-tick-height};
  width: var(--tick-width);
  height: var(--tick-height);
}

.tc__chart__date-tick {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.recharts-yAxis .tc__chart__tick-container {
  transform: translate(calc(-1 * var(--tick-width)), calc(-1 * var(--tick-height) / 2));
}

.tc__chart__text-tick-container {
  $text-tick-width: 40px;
  $text-tick-height: 16px;

  --tick-width: #{$text-tick-width};
  --tick-height: #{$text-tick-height};
  width: var(--tick-width);
  height: var(--tick-height);
}
