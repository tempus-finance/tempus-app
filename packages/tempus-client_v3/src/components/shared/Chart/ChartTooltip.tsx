import { ReactElement, ReactNode } from 'react';
import { TooltipProps } from 'recharts';
import { NameType, ValueType } from 'recharts/types/component/DefaultTooltipContent';
import Tooltip from '../Tooltip';

interface ChartTooltipProps<X extends NameType, Y extends ValueType> {
  tooltipContent: (x: X, y: Y) => ReactNode;
}

function ChartTooltip<X extends NameType, Y extends ValueType>(
  props: ChartTooltipProps<X, Y> & TooltipProps<Y, X>,
): ReactElement<ChartTooltipProps<X, Y> & TooltipProps<Y, X>> {
  const { tooltipContent, coordinate, viewBox, payload } = props;
  return (
    <div className="tc__chart__tooltip-chart">
      <Tooltip
        open
        placement={
          coordinate && coordinate.x && viewBox && viewBox.width && coordinate.x / viewBox.width > 0.5
            ? 'left'
            : 'right'
        }
      >
        {payload && payload[0] !== undefined && tooltipContent(payload[0].payload.x, payload[0].payload.y)}
      </Tooltip>
    </div>
  );
}

export default ChartTooltip;
