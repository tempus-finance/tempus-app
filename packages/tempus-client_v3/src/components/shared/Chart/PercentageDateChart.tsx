import { FC, ReactNode } from 'react';
import { DecimalUtils } from 'tempus-core-services';
import { ChartDataPoint, ChartSizeProps } from './Chart';
import DateChart from './DateChart';

interface PercentageDateChartProps {
  data: ChartDataPoint<Date, number>[];
  topPercentageProjected?: number;
  tooltipContent?: (x: Date, y: number) => ReactNode;
}

const PercentageDateChart: FC<PercentageDateChartProps & ChartSizeProps> = props => {
  const { data, width, height, tooltipContent, topPercentageProjected } = props;

  return (
    <DateChart
      data={data}
      width={width}
      height={height}
      tooltipContent={tooltipContent}
      yTickFormatter={DecimalUtils.formatPercentage}
      topPercentageProjected={topPercentageProjected}
    />
  );
};

export default PercentageDateChart;
