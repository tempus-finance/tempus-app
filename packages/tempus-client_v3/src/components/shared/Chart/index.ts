import BalanceChartTooltipContent from './BalanceChartTooltipContent';
import Chart from './Chart';
import ChartDot from './ChartDot';
import DateChart from './DateChart';
import PercentageDateChart from './PercentageDateChart';

export default Chart;
export { BalanceChartTooltipContent, ChartDot, DateChart, PercentageDateChart };
