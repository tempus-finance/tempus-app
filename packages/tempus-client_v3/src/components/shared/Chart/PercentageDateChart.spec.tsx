import { render } from '@testing-library/react';
import { ChartSizeProps } from './Chart';
import PercentageDateChart, { PercentageDateChartProps } from './PercentageDateChart';

const data = [...Array(20).keys()].map(value => ({
  x: new Date(2022, 3, 1 + value),
  y: (value * value) / 10000,
}));

const defaultProps: PercentageDateChartProps & ChartSizeProps = {
  width: 500,
  height: 300,
};

const subject = (props: PercentageDateChartProps & ChartSizeProps) => render(<PercentageDateChart {...props} />);

describe('PercentageDateChart', () => {
  const originalConsoleWarn = console.warn;

  beforeEach(() => {
    // `render` doesn't properly handle percentage values for chart's width/height, so ignore warnings
    console.warn = jest.fn();
  });

  afterEach(() => {
    console.warn = originalConsoleWarn;
  });

  it('renders a yield chart', () => {
    const { container } = subject({ ...defaultProps, data });

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
