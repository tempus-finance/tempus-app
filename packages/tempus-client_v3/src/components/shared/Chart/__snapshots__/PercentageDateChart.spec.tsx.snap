// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentageDateChart renders a yield chart 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 500px; height: 300px;"
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 500px; height: 300px;"
    >
      <svg
        class="recharts-surface"
        height="300"
        version="1.1"
        viewBox="0 0 500 300"
        width="500"
      >
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="225"
              width="435"
              x="60"
              y="5"
            />
          </clippath>
        </defs>
        <defs>
          <lineargradient
            id="color"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          >
            <stop
              offset="67%"
              stop-color="#129CA4"
              stop-opacity="0.2"
            />
            <stop
              offset="100%"
              stop-color="#129CA4"
              stop-opacity="0"
            />
          </lineargradient>
          <pattern
            height="4"
            id="pattern"
            patternTransform="rotate(45 0 0)"
            patternUnits="userSpaceOnUse"
            width="4"
          >
            <rect
              fill="#2B6E84"
              height="100%"
              opacity="0.2"
              width="100%"
              x="0"
              y="0"
            />
            <line
              opacity="0.3"
              stroke="#000000"
              stroke-width="2px"
              y2="4"
            />
          </pattern>
        </defs>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            orientation="bottom"
            stroke="#666"
            type="number"
            width="435"
            x="60"
            x1="60"
            x2="495"
            y="230"
            y1="230"
            y2="230"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="60"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    01
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="82.89473684210526"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    02
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="105.78947368421052"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    03
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="128.68421052631578"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    04
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="151.57894736842104"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    05
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="174.4736842105263"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    06
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="197.36842105263156"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    07
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="220.26315789473682"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    08
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="243.15789473684208"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    09
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="266.05263157894734"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    10
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="288.9473684210526"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    11
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="311.84210526315786"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    12
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="334.7368421052631"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    13
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="357.63157894736844"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    14
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="380.52631578947364"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    15
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="403.42105263157896"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    16
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="426.31578947368416"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    17
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="449.2105263157895"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    18
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="472.1052631578947"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    19
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__date-tick-container"
                x="495"
                y="238"
              >
                <div
                  class="tc__chart__date-tick"
                >
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
                  >
                    20
                  </div>
                  <div
                    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                  >
                    Apr
                  </div>
                </div>
              </foreignobject>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__text-tick-container"
                x="52"
                y="230"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  0.00%
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__text-tick-container"
                x="52"
                y="167.6731301939058"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  1.00%
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__text-tick-container"
                x="52"
                y="105.34626038781163"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  2.00%
                </div>
              </foreignobject>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <foreignobject
                class="tc__chart__tick-container tc__chart__text-tick-container"
                x="52"
                y="43.01939058171747"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  3.00%
                </div>
              </foreignobject>
            </g>
          </g>
        </g>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="225"
              offset="[object Object]"
              stroke="#F3F7F9"
              width="435"
              x="60"
              x1="60"
              x2="495"
              y="5"
              y1="230"
              y2="230"
            />
            <line
              fill="none"
              height="225"
              offset="[object Object]"
              stroke="#F3F7F9"
              width="435"
              x="60"
              x1="60"
              x2="495"
              y="5"
              y1="167.6731301939058"
              y2="167.6731301939058"
            />
            <line
              fill="none"
              height="225"
              offset="[object Object]"
              stroke="#F3F7F9"
              width="435"
              x="60"
              x1="60"
              x2="495"
              y="5"
              y1="105.34626038781163"
              y2="105.34626038781163"
            />
            <line
              fill="none"
              height="225"
              offset="[object Object]"
              stroke="#F3F7F9"
              width="435"
              x="60"
              x1="60"
              x2="495"
              y="5"
              y1="43.01939058171747"
              y2="43.01939058171747"
            />
            <line
              fill="none"
              height="225"
              offset="[object Object]"
              stroke="#F3F7F9"
              width="435"
              x="60"
              x1="60"
              x2="495"
              y="5"
              y1="5"
              y2="5"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-3"
              >
                <rect
                  height="232"
                  width="0"
                  x="60"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-3)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M60,230C67.63157894736842,229.89612188365652,75.26315789473684,229.79224376731304,82.89473684210526,229.3767313019391C90.52631578947368,228.96121883656514,98.1578947368421,228.33795013850417,105.78947368421052,227.50692520775624C113.42105263157895,226.6759002770083,121.05263157894736,225.63711911357342,128.68421052631578,224.39058171745154C136.3157894736842,223.14404432132966,143.94736842105263,221.6897506925208,151.57894736842104,220.02770083102493C159.21052631578945,218.36565096952907,166.8421052631579,216.49584487534628,174.4736842105263,214.41828254847647C182.1052631578947,212.34072022160666,189.73684210526315,210.05540166204986,197.36842105263156,207.5623268698061C204.99999999999997,205.06925207756234,212.6315789473684,202.36842105263156,220.26315789473682,199.45983379501385C227.89473684210523,196.55124653739614,235.52631578947367,193.43490304709144,243.15789473684208,190.11080332409975C250.7894736842105,186.78670360110806,258.4210526315789,183.2548476454294,266.05263157894734,179.51523545706374C273.6842105263158,175.77562326869807,281.31578947368416,171.8282548476454,288.9473684210526,167.6731301939058C296.57894736842104,163.51800554016617,304.2105263157894,159.15512465373962,311.84210526315786,154.58448753462605C319.4736842105263,150.01385041551248,327.1052631578947,145.23545706371192,334.7368421052631,140.2493074792244C342.36842105263156,135.26315789473685,350,130.06925207756237,357.63157894736844,124.66759002770084C365.2631578947368,119.26592797783935,372.89473684210526,113.65650969529084,380.52631578947364,107.8393351800554C388.1578947368421,102.02216066481994,395.7894736842105,95.99722991689754,403.42105263157896,89.76454293628811C411.05263157894734,83.53185595567871,418.6842105263158,77.09141274238225,426.31578947368416,70.44321329639888C433.9473684210526,63.795013850415486,441.57894736842104,56.93905817174519,449.2105263157895,49.87534626038783C456.84210526315786,42.811634349030506,464.4736842105263,35.540166204986136,472.1052631578947,28.06094182825485C479.7368421052631,20.58171745152353,487.36842105263156,12.790858725761764,495,5L495,230C487.36842105263156,230,479.7368421052631,230,472.1052631578947,230C464.4736842105263,230,456.84210526315786,230,449.2105263157895,230C441.57894736842104,230,433.9473684210526,230,426.31578947368416,230C418.6842105263158,230,411.05263157894734,230,403.42105263157896,230C395.7894736842105,230,388.1578947368421,230,380.52631578947364,230C372.89473684210526,230,365.2631578947368,230,357.63157894736844,230C350,230,342.36842105263156,230,334.7368421052631,230C327.1052631578947,230,319.4736842105263,230,311.84210526315786,230C304.2105263157894,230,296.57894736842104,230,288.9473684210526,230C281.31578947368416,230,273.6842105263158,230,266.05263157894734,230C258.4210526315789,230,250.7894736842105,230,243.15789473684208,230C235.52631578947367,230,227.89473684210523,230,220.26315789473682,230C212.6315789473684,230,204.99999999999997,230,197.36842105263156,230C189.73684210526315,230,182.1052631578947,230,174.4736842105263,230C166.8421052631579,230,159.21052631578945,230,151.57894736842104,230C143.94736842105263,230,136.3157894736842,230,128.68421052631578,230C121.05263157894736,230,113.42105263157895,230,105.78947368421052,230C98.1578947368421,230,90.52631578947368,230,82.89473684210526,230C75.26315789473684,230,67.63157894736842,230,60,230Z"
                  fill="url(#color)"
                  fill-opacity="0.6"
                  height="225"
                  points="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                  stroke="none"
                  stroke-width="2"
                  type="monotone"
                  width="435"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M60,230C67.63157894736842,229.89612188365652,75.26315789473684,229.79224376731304,82.89473684210526,229.3767313019391C90.52631578947368,228.96121883656514,98.1578947368421,228.33795013850417,105.78947368421052,227.50692520775624C113.42105263157895,226.6759002770083,121.05263157894736,225.63711911357342,128.68421052631578,224.39058171745154C136.3157894736842,223.14404432132966,143.94736842105263,221.6897506925208,151.57894736842104,220.02770083102493C159.21052631578945,218.36565096952907,166.8421052631579,216.49584487534628,174.4736842105263,214.41828254847647C182.1052631578947,212.34072022160666,189.73684210526315,210.05540166204986,197.36842105263156,207.5623268698061C204.99999999999997,205.06925207756234,212.6315789473684,202.36842105263156,220.26315789473682,199.45983379501385C227.89473684210523,196.55124653739614,235.52631578947367,193.43490304709144,243.15789473684208,190.11080332409975C250.7894736842105,186.78670360110806,258.4210526315789,183.2548476454294,266.05263157894734,179.51523545706374C273.6842105263158,175.77562326869807,281.31578947368416,171.8282548476454,288.9473684210526,167.6731301939058C296.57894736842104,163.51800554016617,304.2105263157894,159.15512465373962,311.84210526315786,154.58448753462605C319.4736842105263,150.01385041551248,327.1052631578947,145.23545706371192,334.7368421052631,140.2493074792244C342.36842105263156,135.26315789473685,350,130.06925207756237,357.63157894736844,124.66759002770084C365.2631578947368,119.26592797783935,372.89473684210526,113.65650969529084,380.52631578947364,107.8393351800554C388.1578947368421,102.02216066481994,395.7894736842105,95.99722991689754,403.42105263157896,89.76454293628811C411.05263157894734,83.53185595567871,418.6842105263158,77.09141274238225,426.31578947368416,70.44321329639888C433.9473684210526,63.795013850415486,441.57894736842104,56.93905817174519,449.2105263157895,49.87534626038783C456.84210526315786,42.811634349030506,464.4736842105263,35.540166204986136,472.1052631578947,28.06094182825485C479.7368421052631,20.58171745152353,487.36842105263156,12.790858725761764,495,5"
                  fill="none"
                  fill-opacity="0.6"
                  height="225"
                  points="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                  stroke="#2B6E84"
                  stroke-width="2"
                  type="monotone"
                  width="435"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
      <div
        class="recharts-tooltip-wrapper"
        style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="tc__chart__tooltip-chart"
        >
          <div
            class="tc__tooltip tc__tooltip__right"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
